src/lib/services/proposalService.ts

```diff
@@ -252,3 +262,55 @@ export const useProposalTypes = (): UseQueryResult<ProposalType[]> => {
   );
   return useQuery(["proposal_types", indexerGraphClient], queryFn);
 };
+
+interface DepositParamsReturn extends Omit<DepositParams, "min_deposit"> {
+  min_deposit: {
+    amount: string;
+    denom: string;
+    formattedAmount: string;
+    formattedDenom: string;
+    formattedToken: string;
+  };
+}
+
+export const useGovParams = (): UseQueryResult<{
+  depositParams: DepositParamsReturn;
+  uploadAccess: UploadAccess;
+}> => {
+  const lcdEndpoint = useLCDEndpoint();
+  const queryFn = useCallback(() => {
+    return Promise.all([
+      fetchGovDepositParams(lcdEndpoint),
+      fetchGovUploadAccessParams(lcdEndpoint),
+    ]).then<{
+      depositParams: DepositParamsReturn;
+      uploadAccess: UploadAccess;
+    }>((params) => {
+      const minDepositParam = params[0].min_deposit[0];
+      const [minDepositAmount, minDepositDenom] = [
+        demicrofy(minDepositParam.amount as U<Token>).toFixed(),
+        getTokenLabel(minDepositParam.denom),
+      ];
+      return {
+        depositParams: {
+          ...params[0],
+          min_deposit: {
+            ...minDepositParam,
+            formattedAmount: minDepositAmount,
+            formattedDenom: minDepositDenom,
+            formattedToken: formatBalanceWithDenom({
+              coin: minDepositParam,
+              precision: 6,
+              decimalPoints: 2,
+            }),
+          },
+        },
+        uploadAccess: params[1],
+      };
+    });
+  }, [lcdEndpoint]);
+
+  return useQuery(["gov_params", lcdEndpoint], queryFn, {
+    keepPreviousData: true,
```

Comment:
can we omit keepPreviousData?

Reply:
I think it does make sense to cache this query since chain params are almost static